src/index/IndexImpl.Text.cpp

```diff
@@ -833,16 +887,37 @@ Index::WordEntityPostings IndexImpl::getWordPostingsForTerm(
     return wep;
   }
   const auto& tbmd = optionalTbmd.value().tbmd_;
-  wep = readWordCl(tbmd);
+  wep = readWordClWep(tbmd);
   if (optionalTbmd.value().hasToBeFiltered_) {
-    wep = FTSAlgorithms::filterByRange(optionalTbmd.value().idRange_, wep);
+    wep = FTSAlgorithms::filterByRangeWep(optionalTbmd.value().idRange_, wep);
   }
   LOG(DEBUG) << "Word postings for term: " << term
              << ": cids: " << wep.cids_.size() << " scores "
              << wep.scores_.size() << '\n';
   return wep;
 }
 
+// _____________________________________________________________________________
+IdTable IndexImpl::getWordPostingsForTerm(
+    const string& term,
+    const ad_utility::AllocatorWithLimit<Id>& allocator) const {
+  LOG(DEBUG) << "Getting word postings for term: " << term << '\n';
+  IdTable idTable{allocator};
+  auto optionalTbmd = getTextBlockMetadataForWordOrPrefix(term);
+  if (!optionalTbmd.has_value()) {
+    return idTable;
+  }
+  const auto& tbmd = optionalTbmd.value().tbmd_;
+  idTable = readWordCl(tbmd, allocator);
+  if (optionalTbmd.value().hasToBeFiltered_) {
+    idTable =
+        FTSAlgorithms::filterByRange(optionalTbmd.value().idRange_, idTable);
+  }
```

Comment:
You can probably apply the filtering in place (I'll have to look at the function for that).